flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliChangelogResultView.java

```diff
@@ -85,16 +94,6 @@ public CliChangelogResultView(CliClient client, ResultDescriptor resultDescripto
         return Arrays.copyOfRange(resultRow, 1, resultRow.length);
     }
 
-    @Override
-    protected int computeColumnWidth(int idx) {
```

Comment:
To optimize changelog table result view ,I think  computeColumnWIdth with following code will be better.  

```
protected int computeColumnWidth(int idx) {
     PrintUtils.columnWidthByType(column, ....)
}
```

Another benefit is that  constructor of CliResultView is no need to change, subclass of CliResultView can implement computeColumnWIdth  differently.

Reply:
Hi @pensz .

The PR is 200 lines indeed. About half of it is documentation though, and the previous behavior of column width was implemented differently for all 3 modes, so one factor that made the PR grow is my change to make them all depend on the same logic.

The `computeColumnWidth(int idx)` you refer to would be called for every row, which would be marginally slower I think. In this PR the width are now initialized only once, in the constructor, which seems a natural place for performing initialization. 

IMHO, adding methods like `protected int computeColumnWidth(int idx)` adds complexity, since it contributes to making  `CliResultView` more different than the tableau mode, and it adds a degree of freedom to the code which might not add that much value, since using the same column width policy everywhere increase the user experience I think.

Reply:
I second @sv3ndk 's point here. `computeColumnWidth` is not a good design before, because `tableau` mode doesn't use it. `ResultDescriptor` is a good place to put the initialized column widths and make all modes to have the same getting column width logic.